src/utils/embedded/embedded.provider.tsx

```diff
@@ -1157,45 +1145,49 @@ export function EmbeddedProvider({ children }: EmbeddedProviderProps) {
 
       const supabase = await getSupabaseClient();
 
-      // Retrieves the current LOCAL session:
-      const { data, error } = await supabase.auth.getSession();
-      const cachedUser = data?.session?.user;
+      let isInitialAuthEventDispatched = false;
 
-      // Send the initial state for the SDK button ASAP:
+      const {
+        data: { subscription },
+      } = supabase.auth.onAuthStateChange(async (_event, session) => {
```

Comment:
I've changed the session refresh logic a bit and moved the checks to see if the session is complete here.